plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -176,32 +176,43 @@ public static void installAppEngineFacet(IFacetedProject facetedProject,
 
     try {
       if (facetedProject.hasProjectFacet(FACET)) {
+        // nothing to do, move along
         return;
       }
 
+      /*
+       * So the project does not have the App Engine Standard facet. We first try our AES JRE7 and
+       * JRE8 detectors: these detectors look for appengine-web.xml and either add or change the
```

Comment:
Order doesn't matter: they both parse the content and then check if the `<runtime>` element matches their requirement, and return if not.